backend/src/main/java/de/bund/digitalservice/ris/norms/application/validator/Validator.java

```diff
@@ -0,0 +1,15 @@
+package de.bund.digitalservice.ris.norms.application.validator;
+
+import de.bund.digitalservice.ris.norms.utils.exceptions.ValidationException;
+
+/** Interface to be implemented by any custom validator. */
+public interface Validator {
+
+  /**
+   * Validates the passed data within the custom implementation.
+   *
+   * @param args - variable-sized input parameters.
+   * @throws ValidationException whenever a validation step fails
+   */
+  void validate(Object... args) throws ValidationException;
```

Comment:
Can we use a generic instead of `Object...` here? That would allow us to use the type checking to check if our validator is called correctly and we do not need to rely on runtime type checking to ensure that the validator is called correctly. (And additionally it would be simpler to understand what each parameter is supposed to be)

Reply:
That was my first implementation but we still need to have a variable size of input parameters. Now we have the `AmendingLawValidator` only passing a `Norm` object and the `SingleModValidator` passing both a `Norm` object as well as a `Mod` object. We would need to use, for now, two generics for the `SingleModValidator` but the `AmendingLawValidator` only needs one

Reply:
Could we use something like an `Arguments` sub-class within the validator to bundle `Norm` and `Mod` for this case? (So similar to the `Query´ or `Command` things) I would really like to have this type safety here as it seems very simple to use the validators wrongly otherwise.

Reply:
Sure we could but then the `ValidationService`... if needed, would need to be re-worked because calling the same validate method is now working because of the varargs of the input parameters.